stdlib/public/core/MemoryLayout.swift

```diff
@@ -162,4 +162,41 @@ extension MemoryLayout {
   public static func alignment(ofValue value: T) -> Int {
     return MemoryLayout.alignment
   }
+
+  /// Returns the offset of an inline stored property of `T` within the
+  /// in-memory representation of `T`.
+  ///
+  /// If the given `key` refers to inline storage within the
```

Comment:
I think this initial paragraph could focus on the happy path: "If the given `key` refers to inline, directly addressable storage within the in-memory representation of `T`, the the return value…". After the example you have below, with `root` and `value`, we could use your description here of what might disqualify a property from being measured in this way, with an example type like:

```swift
struct ProductCategory {
    var name: String
    var updateCounter: Int
    var products: [Product] {
        didSet { updateCounter += 1 }
    }
    var productCount: Int {
        return products.count
    }
    var someAbstractedProperty: () -> () // I can't think of a good example here
}
```

And then we could point out the valid and invalid properties either in comments or in text. If I'm understanding this right, `updateCounter` and `name` would be safe, but accessing properties of `name` wouldn't—I don't think I included a non-inline property. What do you think?